app-misc/nnn/nnn-4.7.ebuild

```
@@ -0,0 +1,85 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit bash-completion-r1 flag-o-matic toolchain-funcs xdg-utils
+
+DESCRIPTION="The missing terminal file browser for X"
+HOMEPAGE="https://github.com/jarun/nnn"
+SRC_URI="https://github.com/jarun/nnn/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="BSD-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc64 ~x86"
+IUSE="+readline pcre 8contexts icons nerdfonts emoji qsort"
+
+DEPEND="sys-libs/ncurses:=
+	pcre? ( dev-libs/libpcre )
+	readline? ( sys-libs/readline:= )
+	elibc_musl? ( sys-libs/fts-standalone )"
+RDEPEND="${DEPEND}"
+REQUIRED_USE="?? ( icons nerdfonts emoji )"
+
+src_prepare() {
+	default
+	tc-export CC
+	use elibc_musl && append-flags "-lfts"
+	sed -i -e '/install: all/install:/' Makefile || die "sed failed"
+}
+
+src_compile() {
+	emake \
+		O_NORL=$(usex readline 0 1) \
+		O_PCRE=$(usex pcre 1 0) \
+		O_CTX8=$(usex 8contexts 1 0) \
+		O_ICONS=$(usex icons 1 0) \
+		O_NERD=$(usex nerdfonts 1 0) \
+		O_EMOJI=$(usex emoji 1 0) \
+		O_QSORT=$(usex qsort 1 0) \
+		|| die "emake failed"
```

```suggestion
	emake \
		O_NORL=$(usex readline 0 1) \
		O_PCRE=$(usex pcre 1 0) \
		O_CTX8=$(usex 8contexts 1 0) \
		O_ICONS=$(usex icons 1 0) \
		O_NERD=$(usex nerdfonts 1 0) \
		O_EMOJI=$(usex emoji 1 0) \
		O_QSORT=$(usex qsort 1 0) \
		|| die "emake failed"
```
`emake` does not need to `|| die` (since it handled that one on its own already), 
but an array form here would be much more cleaner.

---

I'm not sure if I understood you correctly. Please take another look.